### ISSUE TITLE
Support scroll attributes in Typescript JSX

### ISSUE THREAD

**Issue by Koxiaet** [ryansolid/solid#213]
Currently Solid's JSX Typescript implementation does not support the `scrollLeft` or `scrollTop` properties, and it produces an error.

**Comment by ryansolid** [comment]
Hmm.. I never had intended to support them period (not that I can feasibly stop support props given support for custom elements etc). The challenge with them is they don't work properly if set before connected to DOM. It makes the behavior inconsistent and confusing. Technically it works if updated after the fact but never as the initial state. I was trying to see if there was some precedence for supporting these properties declaratively, React, Vue, Svelte, Preact, Inferno all seem to not support this and recommend imperative APIs.

This takes more than updating the types and I very much dislike the inconsistency of solutions. Special compilations with delay are weird especially as this delay should only happen during initialization. I could post a warning somewhere I suppose, But I'm thinking it's reasonable to not support these properties at this time. I'm open to thoughts here. Am I the only one who thinks it is very confusing that these properties would not work on first render?

**Comment by Koxiaet** [comment]
I didn't realize they had this behaviour, I guess it doesn't really make sense to support it then. Thanks for the explanation.

**Comment by ryansolid** [comment]
I mean to be fair it has to do with render timing. Most declarative libraries build up the elements before they attach them. On the properties behavior makes complete sense. I think it is kind of cool to do stuff like that declaratively. So I actually went ahead and tried variations and did the research. I didn't know 100% how it would work out until I tried. So thank you for that. I'm not opposed to reconsidering this in the future.
